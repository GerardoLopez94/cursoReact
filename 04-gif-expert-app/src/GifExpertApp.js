import React from 'react';

const GifExpertApp  = () =>{

    const categories = ['One Punch', 'Samurai X','Dragom Ball'];

    return (
        <>
            <h2>GifExpertApp</h2>
            <hr />
            <ol>
                {
                    categories.map( category=> {
                        return <li key= {category}>{category}</li>
                    })
                }
            </ol>
        </>
    )
}

export { GifExpertApp };