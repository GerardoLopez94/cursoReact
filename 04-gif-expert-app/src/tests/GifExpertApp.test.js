import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import { GifExpertApp } from '../GifExpertApp';

describe('pruebas en <GifExpertApp />', () =>{

    test('debe de mostrar correctamente', () =>{

        const wrapper = shallow( <GifExpertApp /> );
        
        expect(wrapper).toMatchSnapshot();
    });

    test('debe demostrar una lista de categorias', () =>{
        const categories = ['One Punch', 'Dragom Ball'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })

});