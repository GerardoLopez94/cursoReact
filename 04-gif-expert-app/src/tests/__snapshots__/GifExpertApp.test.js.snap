// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp /> debe de mostrar correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`pruebas en <GifExpertApp /> debe demostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragom Ball"
      key="Dragom Ball"
    />
  </ol>
</Fragment>
`;
